components/items/ItemsGrid/ItemsGrid.vue

```diff
@@ -287,4 +289,40 @@ const getSkeletonVariant = (slotProps) => {
   }
   return 'primary'
 }
+
+const { urlPrefix } = usePrefix()
+const { isAssetHub } = useIsChain(urlPrefix)
+
+const processOnchainData = useDebounceFn(async () => {
+  items.value = await Promise.all(items.value.map(async (item) => {
+    if (item.sn && !isTokenEntity(item)) {
+      const tokenData = await fetchOdaToken(urlPrefix.value, item.collection.id, item.sn)
+
+      if (tokenData.metadata && tokenData.metadata_uri) {
+        return {
+          ...item,
+          name: tokenData.metadata?.name || item.meta.name,
+          meta: {
+            ...item.meta,
+            name: tokenData.metadata?.name || item.meta.name,
+            id: tokenData.metadata?.image || item.meta.id,
+            image: tokenData.metadata?.image || item.meta.image,
+            animationUrl: tokenData.metadata?.animation_url || item.meta.animationUrl,
+          },
+        }
+      }
+    }
+
+    return {
+      ...item,
+      name: item.collection.name,
+    }
+  }))
+}, 500)
+
+watch(() => items.value.length, () => {
+  if (isAssetHub.value && items.value.length && props.fetchOnchainData) {
+    processOnchainData()
+  }
+}, { immediate: true })
```

Comment:
Only the new added items need to be fetched, or it will cause duplicating metadata fetch for the same nft item.